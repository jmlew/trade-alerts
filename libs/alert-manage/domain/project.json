{
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/alert-manage/domain/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/alert-manage/domain/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/alert-manage/domain"],
      "options": {
        "jestConfig": "libs/alert-manage/domain/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": ["type:domain-logic", "domain:alert-manage"]
}
