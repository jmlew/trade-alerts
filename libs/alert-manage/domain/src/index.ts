// All domain entities exposed to other libs.
export * from './lib/entities/alert-manager-api.enum';
export * from './lib/entities/alert-manager-data.enum';
export * from './lib/entities/alert-manager-data.model';

// Expose only the facades which encapsulates all data services.
export * from './lib/application/alert-manager.facade';
