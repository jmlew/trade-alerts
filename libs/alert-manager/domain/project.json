{
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/alert-manager/domain/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/alert-manager/domain/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/alert-manager/domain"],
      "options": {
        "jestConfig": "libs/alert-manager/domain/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": ["type:domain-logic", "domain:alert-manager"]
}
