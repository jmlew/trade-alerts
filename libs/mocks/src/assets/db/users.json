{
  "data": [
    {
      "id": 1,
      "firstName": "John",
      "lastName": "Smith",
      "email": "john@sampleuser.com",
      "createdAt": "2018-07-16T07:32:20.421Z"
    },
    {
      "id": 2,
      "firstName": "Fred",
      "lastName": "Jackson",
      "email": "fred@sampleuser.com",
      "createdAt": "2018-07-16T07:32:20.421Z"
    },
    {
      "id": 3,
      "firstName": "Sam",
      "lastName": "Harris",
      "email": "sam@sampleuser.com",
      "createdAt": "2018-07-18T07:32:20.421Z"
    }
  ]
}
