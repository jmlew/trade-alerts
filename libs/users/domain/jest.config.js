module.exports = {
  displayName: 'users-domain',
  preset: '../../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': 'babel-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../../coverage/libs/users/domain',
};
