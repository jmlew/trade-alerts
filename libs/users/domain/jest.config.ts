/* eslint-disable */
export default {
  displayName: 'users-domain',
  preset: '../../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': ['babel-jest', { presets: ['@nrwl/react/babel'] }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../../coverage/libs/users/domain',
};
