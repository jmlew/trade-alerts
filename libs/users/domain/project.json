{
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/users/domain/src",
  "projectType": "library",
  "tags": ["domain:users", "type:domain-logic"],
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/users/domain/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/users/domain"],
      "options": {
        "jestConfig": "libs/users/domain/jest.config.ts",
        "passWithNoTests": true
      }
    }
  }
}
