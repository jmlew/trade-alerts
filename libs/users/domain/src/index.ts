export * from './lib/application/user.facade';
export * from './lib/entities/user.model';
export * from './lib/infrastructure/user-api.model';
