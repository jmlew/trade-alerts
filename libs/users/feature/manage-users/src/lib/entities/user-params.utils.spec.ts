import { User, UserDetails, UserRecord } from '@trade-alerts/users/domain';

import { getUserFormParams } from './user-params.utils';

describe('User Feature Entities', () => {
  const userDetails: UserDetails = {
    firstName: 'John',
    lastName: 'Smith',
    email: 'john@sampleuser.com',
  };
  it('Returns user form params from existing user record', () => {
    const userRecord: UserRecord = {
      ...userDetails,
      id: 1,
      createdAt: '2018-07-16T07:32:20.421Z',
      updatedAt: '2018-07-16T07:32:20.421Z',
    };
    const formParams = getUserFormParams(userRecord);
    expect(formParams).toMatchSnapshot();
    expect(formParams).toEqual(userDetails);
  });
  it('Returns user form params from existing user', () => {
    const user: User = {
      ...userDetails,
      id: 1,
    };
    const formParams = getUserFormParams(user);
    expect(formParams).toMatchSnapshot();
    expect(formParams).toEqual(userDetails);
  });
  it('Returns user form params from empty user', () => {
    const empty: UserDetails = {
      firstName: '',
      lastName: '',
      email: '',
    };
    const formParams = getUserFormParams(undefined);
    expect(formParams).toMatchSnapshot();
    expect(formParams).toEqual(empty);
  });
});
