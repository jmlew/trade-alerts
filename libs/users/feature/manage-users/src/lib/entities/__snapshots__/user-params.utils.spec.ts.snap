// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Feature Entities Returns user form params from empty user 1`] = `
Object {
  "email": "",
  "firstName": "",
  "lastName": "",
}
`;

exports[`User Feature Entities Returns user form params from existing user 1`] = `
Object {
  "email": "john@sampleuser.com",
  "firstName": "John",
  "lastName": "Smith",
}
`;

exports[`User Feature Entities Returns user form params from existing user record 1`] = `
Object {
  "email": "john@sampleuser.com",
  "firstName": "John",
  "lastName": "Smith",
}
`;
